css_tasks/task9/js/forms.js

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable max-len */
+/* eslint-disable no-undef */
+
+import {constants} from './constants.js';
+
+const {nameText,
+  namePattern,
+  emailText,
+  emailPattern,
+  passwordText,
+  passwordPattern,
+  messageText} = constants;
+
+const submitLoginForm = (e) => {
+  validateName(e.target, e);
+  validateEmail(e.target, e);
+  validatePassword(e.target, e);
+};
+const submitContactForm = (e) => {
+  validateName(e.target, e);
+  validateEmail(e.target, e);
+  validateMessage(e.target, e);
+};
+
+//  Validate Name Input
+
+const validateName = (formName, e) => {
+  const name = formName.elements.name;
+  const nameValue = name.value;
+  hideWarning(name);
+  if (!namePattern.test(nameValue)) {
+    showWarning(name, nameText);
+    e.preventDefault();
+  }
+};
+
+//  Validate Email Input
+
+const validateEmail = (formName, e) => {
+  const email = formName.elements.email;
+  const emailValue = email.value;
+  hideWarning(email);
+  if (!emailPattern.test(emailValue)) {
+    showWarning(email, emailText);
+    e.preventDefault();
+  }
+};
+
+// Validate Password Input
+
+const validatePassword = (formName, e) => {
+  const password = formName.elements.password;
+  const passwordValue = password.value;
+  hideWarning(password);
+  if (!passwordPattern.test(passwordValue)) {
+    showWarning(password, passwordText);
+    e.preventDefault();
+  }
+};
+// Validate Message Textarea
+
+const validateMessage = (formName, e) => {
```

Comment:
all 3 validation functions can be turned into 1 that takes a validation callback of any form of delegation